src/Compilers/Server/VBCSCompiler/CompilerRequestHandler.cs

```diff
@@ -33,11 +33,16 @@ public RunRequest(string language, string currentDirectory, string tempDirectory
         }
     }
 
-    internal abstract class CompilerServerHost : ICompilerServerHost
+    internal sealed class CompilerServerHost : ICompilerServerHost
     {
-        public abstract IAnalyzerAssemblyLoader AnalyzerAssemblyLoader { get; }
+        public IAnalyzerAssemblyLoader AnalyzerAssemblyLoader { get; } = new ShadowCopyAnalyzerAssemblyLoader(Path.Combine(Path.GetTempPath(), "VBCSCompiler", "AnalyzerAssemblyLoader"));
 
-        public abstract Func<string, MetadataReferenceProperties, PortableExecutableReference> AssemblyReferenceProvider { get; }
+        public static Func<string, MetadataReferenceProperties, PortableExecutableReference> SharedAssemblyReferenceProvider { get; } = (path, properties) => new CachingMetadataReference(path, properties);
```

Comment:
>public [](start = 8, length = 6)

`private`?

Reply:
This is used in `TouchedFileLoggingTests`

---
In reply to: [319601452](https://github.com/dotnet/roslyn/pull/38411#discussion_r319601452) [](ancestors = 319601452)